src/routes/settings/SyncNostrContacts.tsx

```diff
@@ -91,11 +94,29 @@ export function SyncContactsForm() {
 
 export default function SyncNostrContacts() {
     const [state, actions] = useMegaStore();
+    const [loading, setLoading] = createSignal(false);
+    const [error, setError] = createSignal<Error>();
 
     function clearNpub() {
         actions.saveNpub("");
     }
 
+    async function resync() {
+        setError(undefined);
+        setLoading(true);
+        try {
+            if (!PRIMAL_API) throw new Error("PRIMAL_API not set");
```

Comment:
should this be outside the `try` that way it is propagated to the user and not just logged?

Reply:
it's correct that it's inside the try but what's incorrect is that I don't setError in the catch. so nice catch on my lack of catch

Reply:
wait, no it's there github just hid it from me